listenbrainz/mbid_mapping/mapping/mb_metadata_cache.py

```diff
@@ -396,51 +399,223 @@ def config_postgres_join_limit(self, curs):
+        # these CTEs concern release data and we fetch release and cover art data from these. the CTEs do not join to
+        # recording table because that has been considered earlier.
+        #
+        # |   CTE / table        |        purpose             |  last_updated considered    | last_updated ignored
+        # |   release_group_tags |  release group level tags  |  release_group_tag, genre   | release, release_group
+        # |   release_data       |  release name, cover art   |                             | release, release_group
+        # |   release            |                            |  release, release_group     |
+        release_mbids_query = """
+            WITH release_mbids(id) AS (
+                SELECT rel.id
+                  FROM mapping.canonical_release_redirect crr
+                  JOIN release rel
+                    ON crr.release_mbid = rel.gid
+                  JOIN release_group rg
+                    ON rel.release_group = rg.id
+                  JOIN release_group_tag rgt
+                    ON rgt.release_group = rel.release_group
+                  JOIN tag t
+                    ON rgt.tag = t.id
+             LEFT JOIN genre g
+                    ON t.name = g.name
+                 WHERE rgt.last_updated > %(timestamp)s
+                    OR   g.last_updated > %(timestamp)s
+            UNION
+                SELECT rel.id
+                  FROM mapping.canonical_release_redirect crr
+                  JOIN release rel
+                    ON crr.release_mbid = rel.gid
+                  JOIN release_group rg
+                    ON rel.release_group = rg.id
+             LEFT JOIN cover_art_archive.cover_art caa
+                    ON caa.release = rel.id
+             LEFT JOIN cover_art_archive.cover_art_type cat
+                    ON cat.id = caa.id
+                 WHERE  rel.last_updated > %(timestamp)s
+                    OR   rg.last_updated > %(timestamp)s
+                    OR (caa.date_uploaded > %(timestamp)s AND (type_id = 1 OR type_id IS NULL))
+            ) SELECT r.gid
+                FROM recording r
+                JOIN track t
+                  ON t.recording = r.id
+                JOIN medium m
+                  ON m.id = t.medium
+                JOIN release_mbids rm
+                  ON rm.id = m.release
+        """
 
-        conn = self.lb_conn if self.lb_conn is not None else self.mb_conn
         try:
-            with conn.cursor() as curs:
-                query = f"""UPDATE {self.table_name}
-                               SET dirty = 't'
-                             WHERE recording_mbid IN %s
-                                OR %s && artist_mbids
-                                OR release_mbid IN %s
-                        """
-                curs.execute(query, (tuple(recording_mbids), artist_mbids, tuple(release_mbids)))
-                conn.commit()
+            with self.mb_conn.cursor() as curs:
```

Comment:
I find it amazing to see the massive queries above, only to collect one column of data. :laughing: